store/postgres/src/metadata.rs

```diff
@@ -0,0 +1,166 @@
+        vid -> BigInt,
+        id -> Text,
+        kind -> Text,
+        name -> Text,
+        network -> Nullable<Text>,
+        source -> Text,
+        mapping -> Text,
+        templates -> Nullable<Array<Text>>,
+        ethereum_block_hash -> Binary,
+        ethereum_block_number -> Numeric,
+        deployment -> Text,
+        block_range -> Range<Integer>,
+    }
+}
+
+table! {
+    subgraphs.subgraph_manifest (vid) {
+        vid -> BigInt,
+        id -> Text,
+        spec_version -> Text,
+        description -> Nullable<Text>,
+        repository -> Nullable<Text>,
+        schema -> Text,
+        data_sources -> Array<Text>,
+        templates -> Nullable<Array<Text>>,
+        block_range -> Range<Integer>,
+    }
+}
+
+table! {
+    subgraphs.ethereum_contract_data_source (vid) {
+        vid -> BigInt,
+        id -> Text,
+        kind -> Text,
+        name -> Text,
+        network -> Nullable<Text>,
+        source -> Text,
+        mapping -> Text,
+        templates -> Nullable<Array<Text>>,
+        block_range -> Range<Integer>,
+    }
+}
+
+/// Look up the graft point for the given subgraph in the database and
+/// return it
+pub fn deployment_graft(
+    conn: &PgConnection,
+    id: &SubgraphDeploymentId,
+) -> Result<Option<(SubgraphDeploymentId, EthereumBlockPointer)>, StoreError> {
+    use subgraph_deployment as sd;
+
+    if id.is_meta() {
+        // There is no SubgraphDeployment for the metadata subgraph
+        Ok(None)
+    } else {
+        match sd::table
+            .select((sd::graft_base, sd::graft_block_hash, sd::graft_block_number))
+            .filter(sd::id.eq(id.as_str()))
+            .first::<(Option<String>, Option<Vec<u8>>, Option<BigDecimal>)>(conn)?
+        {
+            (None, None, None) => Ok(None),
+            (Some(subgraph), Some(hash), Some(block)) => {
+                let hash = H256::from_slice(hash.as_slice());
+                let block = block.to_u64().expect("block numbers fit into a u64");
+                let subgraph = SubgraphDeploymentId::new(subgraph.clone()).map_err(|_| {
+                    StoreError::Unknown(format_err!(
+                        "the base subgraph for a graft must be a valid subgraph id but is `{}`",
+                        subgraph
+                    ))
+                })?;
+                Ok(Some((subgraph, EthereumBlockPointer::from((hash, block)))))
+            }
+            _ => unreachable!(
+                "graftBlockHash and graftBlockNumber are either both set or neither is set"
+            ),
+        }
+    }
+}
+
+pub fn subgraph_schema(
+    conn: &PgConnection,
+    id: SubgraphDeploymentId,
+) -> Result<Schema, StoreError> {
+    // The subgraph of subgraphs schema is built-in and doesn't have a
+    // SubgraphManifest in the database
+    const SUBGRAPHS_SCHEMA: &str = include_str!("subgraphs.graphql");
+    let res = if id.is_meta() {
+        Schema::parse(SUBGRAPHS_SCHEMA, id)
```

Comment:
I wonder what's cheaper: reparsing this schema every time we access the metadata subgraph, or making it `lazy_static!`, parsing it once, and cloning it every time...

Reply:
Yeah, good point. Ideally, this could return an `Arc<Schema>` It's a little tricky right now, but I think once my other PR gets in, it would be easy to do that and then just keep the metadata subgraph in a `lazy_static!` `Arc`